.github/workflows/phpstan.yml

```diff
@@ -0,0 +1,46 @@
+name: PHPStan
+
+on:
+  pull_request:
+  push:
```

Comment:
don't forget to add branch rules, the same as in phpunit workflow, otherwise it will trigger twice, when creatimg MR without forking

Reply:
So that's the thing: I _had_ to remove them, because otherwise _on my fork_ I could not just have them run on push.

I'd wager this should even be changed for the tests so pushes to private forks already trigger them?

Otherwise I would either have to make PRs on my private fork or this repo even when I just want to test something before exposing it.

Reply:
Btw what's MR?

There's another thing too:
- a run on a `pull_request` is not running on the same commit as I pushed something, it's running on a merged commit with the base branch
- this also means if there are merge conflicts, `pull_request` events are not running because the merge is not possible

To be clear: happy to change it, but first want to figure out what we can do about the scenario I described and if we maybe actually can re-enable all branches for `push` again on tests, too? 🤔

Reply:
Sorry, habit after GitLab calling PRs as MRs 😄

Reply:
> 

In that case, let's push again. Better than leaving branch rules as "blank".

Reply:
> it's running on a merged commit with the base branch

I think it is even better, helping us track possible issues before they appear


> not running because the merge is not possible

Resolve conflicts (either merge target branch, or rebase onto latest commit), test will run and then merge

Reply:
> In that case, let's push again. Better than leaving branch rules as "blank".

Apologies, what do you mean exactly? Are you requesting too I add back the same branches which are in the phpunit.yml?

Reply:
> 

Yes, in fact, I was thinking more to "move" the branches from one to another. But I'm reading more carefully here might this is will not work, better forgot what I told.